python/pyspark/ml/deepspeed/deepspeed_distributor.py

```diff
@@ -0,0 +1,151 @@
+        specify the number of gpus per node and the number of nodes (the same as if running from terminal),
+        as well as specify a deepspeed configuration file.
+
+        Parameters
+        ----------
+        num_gpus: int
+            The number of GPUs to use per node (analagous to num_gpus in deepspeed command).
+
+        nnodes: int
+            The number of nodes that should be used for the run.
+
+        local_mode: bool
+            Whether or not to run the training in a distributed fashion or just locally.
+
+        use_gpu: bool
+            Boolean flag to determine whether to utilize gpus.
+
+        deepspeed_config: Union[Dict[str,Any], str] or None:
+            The configuration file to be used for launching the deepspeed application.
+            If it is a dictionary mapping parameters to values, then we will create the file.
+            If None, deepspeed will fall back to default parameters.
+        """
+        num_processes = num_gpus * nnodes
+        DEEPSPEED_SSL_CONF = "deepspeed.spark.distributor.ignoreSsl"
+        self.deepspeed_config = deepspeed_config
+        super().__init__(num_processes, local_mode, use_gpu, _ssl_conf=DEEPSPEED_SSL_CONF)
+        self.cleanup_deepspeed_conf = False
+
+    @staticmethod
+    def _get_deepspeed_config_path(deepspeed_config) -> str:
+        if isinstance(deepspeed_config, dict):
+            with tempfile.NamedTemporaryFile(mode="w+", delete=False, suffix=".json") as file:
+                json.dump(deepspeed_config, file)
+                return file.name
+        deepspeed_config_path = deepspeed_config
+        # Empty value means the deepspeed will fall back to default settings.
+        if deepspeed_config == None:
+            return ""
+        return deepspeed_config_path
+
+    @staticmethod
+    def _create_torchrun_command(
+        input_params: Dict[str, Any], train_path: str, *args: Any
+    ) -> List[str]:
+        local_mode = input_params["local_mode"]
+        num_processes = input_params["num_processes"]
+        deepspeed_config = input_params["deepspeed_config"]
+        deepspeed_config_path = DeepspeedTorchDistributor._get_deepspeed_config_path(
+            deepspeed_config
+        )
+        torchrun_args, processes_per_node = TorchDistributor._get_torchrun_args(
+            local_mode, num_processes
+        )
+        args_string = list(map(str, args))
+        command_to_run = [
+            sys.executable,
+            "-m",
+            "torch.distributed.run",
+            *torchrun_args,
+            f"--nproc_per_node={processes_per_node}",
+            train_path,
+            *args_string,
+            "-deepspeed",
+            "--deepspeed_config",
+            deepspeed_config_path,
+        ]
+
+        # Don't have the deepspeed_config argument if no path is provided or no parameters set
+        if deepspeed_config_path == "":
+            command_to_run.pop()
+            command_to_run.pop()
```

Comment:
Define command_to_run without the last two list items. If deepspeed_config_path is empty, return command_to_run. Else, append "--deepspeed_config" and deepspeed_config_path and return (can just do like `command_to_run + ["--deepspeed_config", deepspeed_config_path]`